### DISCUSSION TITLE
Support for PWAs?

### DISCUSSION THREAD

**Discussion by Ristovski** [kaleidawave/prism#34] [Q&A] [answered]
Can `prism` work from a static file host? This would also make it suitable for PWAs, but the readme does not go into details on this. How would stuff like `@Page` work?

**Comment by kaleidawave** [comment, marked as answer]
Hey, that's a good question! 

I haven't built a PWA using Prism before. If you build with `prism compile-app --context client` then it should generate a index.html and client side JS in the output folder which could be uploaded to a static host (gh pages, firebase, cf pages, Netlify etc). The `@Page` decorator generates client side routing code so you could build the SPA with client side routing a PWA would need. However you would have to set up the service worker manually and some of that stuff I haven't tested.

Note that Prism is not actively maintained as I am working on a superior replacement. Feel free to try the unique features but I wouldn't recommend putting using it for production :)
